rxnetty/src/main/java/io/reactivex/netty/protocol/http/websocket/WebSocketClient.java

```diff
@@ -80,8 +80,14 @@ public void onNext(final ObservableConnection<T, T> connection) {
                         handler.addHandshakeFinishedListener(new ChannelFutureListener() {
                             @Override
                             public void operationComplete(ChannelFuture future) throws Exception {
-                                originalSubscriber.onNext(connection);
-                                originalSubscriber.onCompleted();
+                                if (future.isCancelled()) {
+                                    originalSubscriber.unsubscribe();
```

Comment:
The below code is sufficient for handling the future completion:

```java
if (future.isSuccess()) {
       originalSubscriber.onNext(connection);
       originalSubscriber.onCompleted();
} else {
        originalSubscriber.onError(future.cause());
}
```

For cancellation, `isSuccess()` returns `false` and `cause()` returns `CancellationException`